### ISSUE TITLE
'GriddedDataset' object is not callable

### ISSUE THREAD

**Issue by briannazawadzki** [MPoL-dev/MPoL#95]
Sometimes the error message `'GriddedDataset' is not callable` appears. It resolves when I start a new job/reboot the kernel.

It can pop up at seemingly random times, but it frequently pops up after running an optimization loop, something like:
```
model.load_state_dict(torch.load("model.pt"))
optimizer = torch.optim.Adam(model.parameters(), lr=config["lr"])
writer = None
loss_val, loss_track = train(model, test4, optimizer, config, device=device, writer=writer)
vis = model.forward()
ground_test4 = utils.packed_cube_to_ground_cube(vis)
```
**if also followed by cross validation**, e.g.,
```
def cross_validate(config):
    test_scores = []

    for k_fold, (train_dset, test_dset) in enumerate(k_fold_datasets):
        rml = precomposed.SimpleNet(coords=coords, nchan=train_dset.nchan)
        optimizer = torch.optim.Adam(rml.parameters(), lr=config["lr"])
        train(rml, train_dset, optimizer, config)
        test_scores.append(test(rml, test_dset))

    test_score = np.sum(np.array(test_scores))
    return test_score, test_scores

cv_score, ind_cv_scores = cross_validate(config)
```
Unfortunately I haven't been able to pinpoint the issue in further detail. This error has also popped up at other times so it's not isolated to one script in particular. I know @trq5014 has also run into this before.

It would be great if we could find where this issue is coming from.

**Comment by jeffjennings** [comment]
I just noticed this error in a script to run mpol that takes some code blocks from the tutorials; it's not an error in the codebase, just a bit of a careless choice of naming in the tutorial. If for example you define `train` as a function, but then write a loop that starts with `for i, (train, test) in enumerate(k_fold_datasets):`, you've now redefined `train`. The tutorial still works because the `train` and `test` functions are defined in a block after the loop. But in a script you may have already defined the `train` and `test` functions (as I did). Probably best to save some people confusion and just change the loop variables in the tutorial.

> Alright, I've made a barebones example that reproduces the error. Here it is in plain text. [GriddedDatasetNotCallable.txt](https://github.com/MPoL-dev/MPoL/files/7393301/GriddedDatasetNotCallable.txt)
> 
> Bizarrely, the error does NOT appear if you remove the following:
> 
> ```
> for i, (train, test) in enumerate(k_fold_datasets):
>     print(i)
> ```
> 
> Obviously, the loop above does essentially nothing. But I was trying to visualize the cross-validation partitioning, making a plot much like the plot with the train and test masks in the [cross-validation tutorial (See cell 9)](https://mpol-dev.github.io/MPoL/ci-tutorials/crossvalidation.html). I assumed that the error was coming from somewhere inside the for loop, but it seems to be the for loop itself. However, we do not see this error in the cross-validation tutorial. I just ran the CV tutorial successfully on my venv so I don't think it's my MPoL installation, and I replaced the dataset in the tutorial with the HD143006 data and it still worked.

**Comment by jeffjennings** [comment]
(In your own script you could alternatively just put the call to `for i, (train, test) in enumerate(k_fold_datasets):` inside a function, like one that makes the k-fold figure in the cross-val tutorial).
